### ISSUE TITLE
Define and export some instance of `EarthParameterSet`

### ISSUE THREAD

**Issue by Datseris** [CliMA/CLIMAParameters.jl#49]
Hi,

I am not sure people should do:
```julia
using CLIMAParameters
struct EarthParameterSet <: AbstractEarthParameterSet end
const param_set = EarthParameterSet()
CLIMAParameters.Planet.grav(param_set)
```

instead of what I'd recommend, which is:
```julia
import CLIMAParameters
set = CLIMAParameters.EARTH_PARAMETER_SET # in capitals because its global constant
CLIMAParameters.grav(set)
```

in the case that they want to use CLIMAParameters to access constants. It just feels a bit weird to have to define a _new type_ to use such an API which in principle is just about obtaining values.

**Comment by charleskawczynski** [comment]
I think something like this would be okay. Users would still be able to subtype from `AbstractParameterSet`.

**Comment by trontrytel** [comment]
Whats the benefit of doing it this way?

**Comment by Datseris** [comment]
> Whats the benefit of doing it this way?

From my own personal perspective: It just feels a bit weird to have to define a new type to use such an API which in principle is just about obtaining values. 

At least, it is definitely not the approach shared by other "constant providing libraries" in Julia. You typically "get the thing from the library" and you don't have to create your own types in the process. It is a more a mental picture kind of thing, not a "saving a 1 line of code" kind of thing. Generating your own structs is a much higher order of complexity for a normal analyst like me that just wants to write some simple code.

**Comment by trontrytel** [comment]
> > Whats the benefit of doing it this way?
> 
> From my own personal perspective: It just feels a bit weird to have to define a new type to use such an API which in principle is just about obtaining values.
> 
> At least, it is definitely not the approach shared by other "constant providing libraries" in Julia. You typically "get the thing from the library" and you don't have to create your own types in the process. It is a more a mental picture kind of thing, not a "saving a 1 line of code" kind of thing. Generating your own structs is a much higher order of complexity for a normal analyst like me that just wants to write some simple code.

Thank you for the explanation. I think you are right. It seems easier to just get a constant this way.

**Comment by glwagner** [comment]
I also agree with @Datseris .

**Comment by Datseris** [comment]
Notice also that you could simplify things even further by changing the source code functions so that they use the default set.

E.g., in the source you have:
```julia
const DEFAULT_EARTH_PARAMETERS = EarthParameterSet()
```

then, when you define the `grav` and all other functions you could do it this way:
```julia
grav(param::ABS = DEFAULT_EARTH_PARAMETERS) = 9.8
```

This way, a user that really only cares about obtaining default values from a well maintained library only has to do:
```julia
import CLIMAParameters
g = CLIMAParameters.grav()
```

**Comment by charleskawczynski** [comment]
> Notice also that you could simplify things even further...

I'll need to think a bit more about this. There's a surprising number of constraints and requirements on this package / layer. I don't see anything wrong with this at first glance.

The only downside is that all parameter set definitions would need to reside in CLIMAParameters, which is not necessarily bad, but it does become slightly more opaque to ClimateMachine driver files--and we've had complaints about driver simplicity / transparency. Regardless, this approach allows to still be overloaded for modifying subsets of parameters, so I don't see any technical limitations.

cc @bischtob, @glwagner, @sandreza and @kpamnany

**Comment by kpamnany** [comment]
First, I agree that there should be a pre-defined parameter set, as described in the OP.

However I'm not sure the default parameter is a good idea. One of the essential requirements for this package is that it should be difficult to end up with different values for a parameter. If you're overriding parameters, e.g.:
```julia
using CLIMAParameters
struct EarthParameterSet <: AbstractEarthParameterSet end
const param_set = EarthParameterSet()
CLIMAParameters.Planet.grav(::EarthParameterSet) = 4.5
```
You can quite easily end up making the mistake of simply calling `grav()` instead of `grav(param_set)` and getting the parameter for `EARTH_PARAMETER_SET` rather than your custom parameter.

**Comment by Datseris** [comment]
> The only downside is that all parameter set definitions would need to reside in CLIMAParameters

I was talking about the existing parameters, which are anyways defined. This does not necessarily put a constraint on some "special" parameters so to speak, that really only make sense to be defined on a externally created `ParameterSet`.

But more importantly, with my proposal the existing interface stays completely unaffected, because due to multiple dispatch, a method `grav()` and a method `grav(param)` are completely independent from a functional perspective.
 
> You can quite easily end up making the mistake of simply calling grav() instead of grav(param_set) and getting the parameter for EARTH_PARAMETER_SET rather than your custom parameter.

I don't think this argument is very convincing against the proposal. First, if you are over-writing parameters, then you anyways want to use your concrete, externally defined instance of `ParameterSet`, why would you ever call `grav()`? Second, I could argue, you can also easily make the mistake of using `grav(param1)` instead of `grav(param2)` which is how the existing interface works. Third, the API is quite clear and such mistakes are 100% on the user to blame, just as much it is a mistake to call `x = rand()` when you'd want `x` to be of type `Float32` instead of the default `Float64`. Because in Julia the standard way to define default behaviors is via multiple dispatch, many functions have no argument and 1 argument versions, the former being the default.

**Comment by bischtob** [comment]
Based on our experience working with this right now, we may want to rethink the interface :).

**Comment by glwagner** [comment]
I agree that users shouldn't have to call `grav(set)` in their scripts and that defaults are appropriate in the case of physical constants for Earth. Some utilities like

```julia
parameters = @planet_parameters begin
    "MercuryParameters"
    gravitational_acceleration = 3.79
    rotation_rate = 1 / 59earth_days
end

model = AtmosphericModel(parameters=parameters, ...)
```

that creates a new parameter type, defines appropriate functions via dispatch, and brings `gravitational_acceleration` into scope might help.

**Comment by bischtob** [comment]
> I agree that users shouldn't have to call `grav(set)` in their scripts and that defaults are appropriate in the case of physical constants for Earth. Some utilities like
> 
> ```julia
> parameters = @planet_parameters begin
>     "MercuryParameters"
>     gravitational_acceleration = 3.79
>     rotation_rate = 1 / 59earth_days
> end
> 
> model = AtmosphericModel(parameters=parameters, ...)
> ```
> 
> that creates a new parameter type, defines appropriate functions via dispatch, and brings `gravitational_acceleration` into scope might help.

I second this kind of approach.

**Comment by sandreza** [comment]
I like @glwagner 's approach here

**Comment by charleskawczynski** [comment]
Closed by #79. (we no longer ship these types, we only define the values and users must define their own types).
